### PR TITLE
fix(directory): enable repo_root_style when truncation_length is zero.

### LINKED ISSUE #3221: Custom repo_root_style does not work with truncation_length = 0
(URL generated by `starship bug-report` gave an HTTP ERROR 400 in the browser, so didn't use that. Sorry.)

#### Current Behavior
Custom `repo_root_style` does not work with `truncation_length = 0`.

#### Expected Behavior
Custom `repo_root_style` should work with `truncation_length = 0`.

#### Additional context/Screenshots
If I remove `truncation_length = 0` from my config or set it to a value different from `0`, the directory which is the repo root correctly shows with the style defined in `repo_root_style`. If I keep `truncation_length = 0` in my config, the directory which is the repo root shows with the same style as the other directories.

According to the documentation on `truncation_length`:

> `0` means no truncation. 

so I'm considering it as a valid value, and that should not interfere with `repo_root_style`.

#### Environment
- Starship version:
    ```
    starship 1.0.0
    branch:
    commit_hash:
    build_time:2021-11-09 21:30:28 +00:00
    build_env:rustc 1.56.1,
    ```
- Shell type: zsh
- Shell version: `zsh 5.8 (x86_64-apple-darwin20.0)`
- Shell plugin manager: Zim
- Terminal emulator: iTerm
- Operating system: macOS 11.6.1

#### Relevant Shell Configuration

```bash
if (( ${+commands[starship]} )) eval "$(starship init zsh)"
```

#### Starship Configuration

```bash
[directory]
truncation_length = 0
truncate_to_repo = false
repo_root_style = 'bold red'
read_only = ' '
```

### PR DESCRIPTION
**Pull request by t-mangoe**
#### Description

I improved the code in order to enable `repo_root_style` when `truncation_length` is zero.

Also, the description in the documentation (`docs/config/README.md`) was incorrect, so I fixed the description.
(`repo_root_style` option is enabled just by setting a value)

#### Motivation and Context

Closes #3221 

#### Screenshots (if appropriate):
スクリーンショット 2022-01-28 23 14 19

#### How Has This Been Tested?

- [x] I have tested using **MacOS** ( <- Third party confirmed)
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: docs/config/README.md [hunk 1]
```diff
@@ -1250,7 +1250,7 @@ This is based on the `~/.config/gcloud/active_config` file and the `~/.config/gc
 | Option            | Default                                                  | Description                                                      |
 | ----------------- | -------------------------------------------------------- | ---------------------------------------------------------------- |
 | `format`          | `'on [$symbol$account(@$domain)(\($region\))]($style) '` | The format for the module.                                       |
-| `symbol`          | `"☁️  "`                                                 | The symbol used before displaying the current GCP profile.       |
+| `symbol`          | `"☁️ "`                                                  | The symbol used before displaying the current GCP profile.       |
```
Comment (davidkna) [inline docs/config/README.md:1253]: Please don't add unrelated changes to the docs.

Comment (t-mangoe) [reply docs/config/README.md]: Thank you for your comment.
This is an unintended change.
I corrected.

**Comment by ccqpein** [comment]
@t-mangoe 

I met the same issue tonight and find this PR before I try to make mine. 

I use macOS Monterey 12.3. I install `t-mangoe:zero_truncation_repo_style` branch and it works on my laptop. 

rust version: `rustc 1.61.0-nightly (3c17c84a3 2022-03-21)`

I think you can mark `I have tested using MacOS` to done in your description. 

cc @davidkna

File: docs/config/README.md [hunk 2]
```diff
@@ -841,20 +841,20 @@ it would have been `nixpkgs/pkgs`.
...
+| `read_only`         | `"🔒"`                                                                                                       | The symbol indicating current directory is read only.                                                                              |
+| `read_only_style`   | `"red"`                                                                                                     | The style for the read only symbol.                                                                                                |
+| `truncation_symbol` | `""`                                                                                                        | The symbol to prefix to truncated paths. eg: "…/"                                                                                  |
+| `repo_root_style`   | `None`                                                                                                      | The style for the root of the git repo when `truncate_to_repo` option is set to false. The default value is equivalent to `style`. |
```
Comment (davidkna) [inline docs/config/README.md:854]: Please revert this. It also works with `truncate_to_repo` enabled.

Comment (t-mangoe) [reply docs/config/README.md]: I reverted README.md

**Comment by t-mangoe** [comment]
@ccqpein
Thank you for your comment!!
OK, I marked `I have tested using MacOS`.

**Review by davidkna — APPROVED** [review]
